### DISCUSSION TITLE
When creating middleware, is there a better option then `event.node.req.url`?

### DISCUSSION THREAD

**Discussion by sasial-dev** [nitrojs/nitro#714] [Q&A]
I'm trying to create auth middleware, (and other than the lack of docs on middleware 😞), I can't seem to find an alternative to `event.node.req.url`. Am I missing anything?

**Comment by misaon** [comment]
`event.req.url` ?

**Reply by sasial-dev** [reply]
In nitro 0.6.0 (h3 0.8.0) `event.req`/`event.res` was deprecated for `event.node.req`/`event.node.res`

**Comment by TheAlexLichter** [comment]
What exactly is your goal/problem? 🤔

**Reply by sasial-dev** [reply]
I'm just wondering if there is a native way to get the URL, as I assume you want to move off the relience of node, but in non-node events you can't get the URL yet!
This is also sorta related to https://github.com/unjs/nitro/issues/715

**Comment by stafyniaksacha** [comment]
You may want to use [h3 utilities](https://github.com/unjs/h3#utilities). 

It seems that `getRequestUrl` does not exist, maybe it should be added there? https://github.com/unjs/h3/blob/main/src/utils/request.ts

**Reply by sasial-dev** [reply]
See #715, getRouterParams are empty in middleware. Not sure if it applies to the others.
